sagan/callbacks.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import os
+
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import wandb
+from tensorflow import keras
+
+from sagan.utils import plot_results
+
+
+class GanMonitor(keras.callbacks.Callback):
+    def __init__(
+        self,
+        val_dataset: tf.data.Dataset,
+        n_samples: int,
+        epoch_interval: int,
+        use_wandb: bool,
+        plot_save_dir: str,
+        num_images_plot: int = 4,
+    ):
+        self.val_images = next(iter(val_dataset))
+        self.n_samples = n_samples
+        self.epoch_interval = epoch_interval
+        self.plot_save_dir = plot_save_dir
+        self.use_wandb = use_wandb
+        self.columns = ["Generated Image"]
```

Comment:
@sayannath A table with a single column has no use as a table. You can log the images directly without logging them as a table. Just use 

```python
for image in images:
        wandb.log({"Generated Images": wandb.Image(image_array, caption="Top: Output, Bottom: Input")}, step=epoch)
```